specreduce/extract.py

```diff
@@ -134,6 +142,12 @@ def __call__(self, image, trace_object, width=5,
             The extracted 1d spectrum with flux expressed in the same
             units as the input image, or u.DN, and pixel units
         """
+        image = image if image is not None else self.image
```

Comment:
not a showstopper or blocker by any means, but there should be a more elegant way to handle this pattern. i'll give it a think.

Reply:
that would definitely be nice. I'm not sure if dataclass has anything built-in. If not, we could implement a simple decorator to parse input arguments and fallback on the set defaults?

Reply:
actually, shouldn't it work to just set the default values to the `self.*` ones rather than `None`? that's effectively what you're doing, anyway.

Reply:
I think that would require a decorator, I don't think `def __call__(self, image=self.image, trace_object=self.trace_object, ...)` is valid syntax, is it?  Or is there some other way to handle defaults?